Check socket and ssl connection exist before closing.

serargz opened the pull request:
As the socket or the ssl connection might not be initialized when
closing, we need to check its existance before doing it.

serargz committed:
Check socket and ssl connection exist before closing.

As the socket or the ssl connection might not be initialized when
closing, we need to check its existance before doing it.

chaffeqa commented:
getting this error as well :+1:

t413 commented:
I'm getting this too. +1

lserman commented:
Also ran into this, I'm wondering though - why does with_connection even rescue from StandardError? Even with this PR, the error message ("The path to your pem file does not exist!") will be swallowed until the retries are over. I think the pem error messages should fail immediately, retries aren't going to help a missing pem.

senekis committed:
Support the notification action: "category" in APNS

Is a new option added in iOS8 that allow users to interact with custom
functionality in the context of a user task.

serargz committed:
Merge pull request #1 from senekis/master

Support the notification action: "category" in APNS
